zhaquirks/tuya/ts0601_trv_sas.py

```diff
@@ -155,18 +178,25 @@ def system_mode_reported(self, value):
             )
             _LOGGER.debug("reported system_mode: off")
 
+    def schedule_state_reported(self, value):
+        """Handle reported schedule state."""
+        if value == 1:
+            _LOGGER.debug("reported schedule state: enabled")
+        else:
+            _LOGGER.debug("reported schedule state: disabled")
```

Comment:
The implementation is updating the `schedule_enabled` attribute when the `system_mode` is updated. Why don't just let the user write the attribute here?
```python
    def schedule_state_reported(self, value):
        """Handle reported schedule state."""

        self._update_attribute(self.attributes_by_name["schedule_enabled"].id, value)

```

Reply:
That will only change the value in the cache. Just making sure that’s known

Reply:
This quirk does not implement schedule programming therefore IMO it should automatically force schedule mode to be off, without involving the user.

Reply:
It's fine to me.

Just to say that the `SASSWEL_SCHEDULE_ENABLE_ATTR: 0` is performed here:
```python
    def map_attribute(self, attribute, value):
        """Map standardized attribute value to dict of manufacturer values."""

        if attribute == "occupied_heating_setpoint":
            # centidegree to decidegree
            return {SASSWEL_HEATING_SETPOINT_ATTR: round(value / 10)}

        if attribute == "system_mode":
            # this quirk does not support programmig modes so we force the schedule mode to be always off
            # more details: https://github.com/zigpy/zha-device-handlers/issues/1815
            if value == self.SystemMode.Off:
                return {SASSWEL_STATE_ATTR: 0, SASSWEL_SCHEDULE_ENABLE_ATTR: 0}
            if value == self.SystemMode.Heat:
                return {SASSWEL_STATE_ATTR: 1, SASSWEL_SCHEDULE_ENABLE_ATTR: 0}
```

and takes place every time that the `system_mode` is written to the `Off` or `Heat` values. I don't know if there can be other values for the `system_mode`.

Reply:
This device has only `Off` and `Heat` modes. Disabling the builtin schedule on each state change assures that it always stays off.